src/ISBootloaderAPP.cpp

```diff
@@ -48,14 +48,24 @@ eImageSignature cISBootloaderAPP::check_is_compatible()
     is_comm_instance_t comm;
     uint8_t buffer[2048];
     is_comm_init(&comm, buffer, sizeof(buffer));
-    int messageSize;
+    int messageSize, n, i, m;
```

Comment:
Curious, but not a problem, why you moved all the temp variables up here, rather that keeping them in their respective scopes?